MAINTAINERS.md

```diff
@@ -77,8 +87,66 @@ Make frequent project releases to the community.
+Individuals accept the nomination by replying, or commenting, for example _"Thank you! I would love to."_
+
+### Public Confirmation
+
+Upon receiving 3 positive maintainer votes, and no vetoes, from other maintainers, and after having confirmed interest with the nominee, the maintainer opens a pull request adding the proposed co-maintainer to MAINTAINERS.md, and perform a final vote similar to the nomination process. If the nomination still has 3 positive votes, and no vetoes, from other maintainers, the nomination is accepted. The positive maintainer voters should approve the pull request to show support.
+
+> _Content from the above nomination._
+> 
+> Here is my +1.
+> 
+> If you support this nomination, please add your +1 or a comment. 
+> 
+> If you do not support this nomination, we invite you to e-mail all/some of the current [project maintainers](https://github.com/opensearch-project/OpenSearch/blob/main/MAINTAINERS.md) in the next 2 weeks. We will keep your feedback private to this group and discuss offline. We will not publicly post the reasons for a decision not to add this co-maintainer, or publicize your name. 
+> 
+> Anyone, including maintainers, can publicly add +1s as requested in the nomination issue if they support the nomination or contact the maintainers in private if they oppose. The pull request stays open for 2 weeks.
+
+### Maintainer Decision
+
+If a positive decision has been reached, the nomination is accepted. If the maintainers cannot reach a positive decision for any reason, the nomination is rejected with _"a decision could not be reached"_ with no further explanation necessary, and the pull request is closed.
+
+### Adding a New Maintainer
+
+The repo admin adjusts the new maintainer’s permissions accordingly, and merges the pull request to MAINTAINERS.md.
+
+## Removing a Maintainer
+
+Removing a maintainer is a disruptive action that the community of maintainers should not undertake lightly. There are several reasons a maintainer might be removed from the project, such as inactivity, or taking actions that negatively impact the project.
+
+### Moving On
+
+There are plenty of reasons that might cause someone to want to take a step back or even a hiatus from a project. Existing maintainers can choose to leave the project at any time, with or without reason, by making a pull request to move themselves to the "Emeritus" section of MAINTAINERS.md, and asking an admin to remove their permissions.
+
+### Inactivity
+
+Maintainer status never expires. If a maintainer becomes inactive for a time (usually several months), the repo admin may revoke maintainer level access to the repository for security reasons. A maintainer can reach out to the repo admin to get their permissions reinstated.
+
+If the repo is left without any maintainers, either by maintainer inactivity or moving on, the repo is considered unmaintained. The repo admin will seek out new maintainers and note the maintenance status in the repo README file.
```

Comment:
```suggestion
If the repo is left without any maintainers, either by maintainer inactivity or moving on, the repo is considered unmaintained. The repo admin will seek out new maintainers and note the maintenance status in the repo README file. This could lead to a repo becoming archived due to lack of maintainers. 
```

Reply:
Archiving repos is a separate topic. I do like this, but I will keep it out of this PR.

Reply:
> But I don't think the wider community should have a chance to publicly discuss this individual's additional commit/maintainer karma -- it should only be up to the existing maintainers to make that decision (and then announce it widely, publicly).

Let's agree to disagree. There's nothing preventing public comment from happening, certainly not a delay of keeping the issue open for a couple of weeks. The text in the PR is inviting people to comment in private, and during that time someone reasonable may offer a reasonable argument to not add a maintainer, and e-mail the maintainers in private. If we just go ahead, they are more likely to manifest their disagreement in public, which I think is net worse.

All that said, if someone else feels strongly otherwise, I'd like to hear from you!

Reply:
> Archiving repos is a separate topic.

Good point. Probably best to add this language after there's a process to put a project in maintenance mode. I've [opened an issue](https://github.com/opensearch-project/.github/issues/68) for this. The Linux Foundation [has some good material](https://www.linuxfoundation.org/tools/winding-down-an-open-source-project/) we can leverage. I'd also be curious to see what @ke4qqq @rbowen think

Reply:
I do feel strongly about it for a couple of reasons.

One is historical bias (ie, "we've always done it that way") from Apache involvement. Admittedly, that's never a great reason.

But I think it goes deeper than that. Having discussion of personnel issues (and that's what this is) in the public record - particularly in a public mailing list, which will be archived, permanently, forever, can open us ("us" = project, company, individuals) up to legal problems, and can follow the person being discussed throughout the rest of their career. For this reason, at Apache at least, we have always held these kinds of conversations private.

The other aspect of this is that it can be deeply humiliating, and career-limiting, to have a public conversation (again, permanently archived) where one's name is suggested for commit/maintainer, and the conclusion is "no". Potential employers will find that conversation forever, and that is hugely unfair (not to mention the above legal consideration) to the person in question.

So, yes, for these reasons, and perhaps others, we should steer all discussion of proposed committers and maintainers to a private list where people are expected to respect confidentiality.

Reply:
https://github.com/opensearch-project/.github/pull/59/commits/6947878111ed664be9225f0a2410a2cfed4be468